system/lib/pthread/pthread_create.c

```diff
@@ -206,6 +223,10 @@ void _emscripten_thread_exit(void* result) {
   // gets shut down during __pthread_tsd_run_dtors.
   emscripten_builtin_free(self->tsd);
   self->tsd = NULL;
+  if (self->stack_owned) {
+    emscripten_builtin_free(self->stack - self->stack_size);
+    self->stack_owned = 0;
+  }
```

Comment:
Nice! I assume this same logic cannot be removed from `_emscripten_thread_free_data`, since that is also used by `pthread_kill` / `$killThread`, right? If so, commit https://github.com/emscripten-core/emscripten/pull/13007/commits/2ac46e2617bc8372fb7d459907628f5c2a3ed6ee within PR #13007 could help with this.

Reply:
Oops, I didn't mean to add this as part of this PR.  Will remove..